### PR TITLE
[metrics] speed up SSIM tests

### PR DESCRIPTION
**Pull request by ydcjeff**
Description: speed up SSIM tests

**before**
Screen Shot 2020-11-15 at 17 40 05

**after**
Screen Shot 2020-11-15 at 17 39 59

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by ydcjeff** [comment]
@vfdev-5 should we use `pytest.mark.parametrize` for `test_ssim()` test part? we are now writing the same code twice with diff parameters.

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @ydcjeff !
I left few comments. As for pytest parameterized, this can be an enhancement for all tests (let's file an issue about that). Currently, to factorize the tests we are implementing `_test_something(params)` and call them from `test_something`. Can we do the same thing here ?

File: tests/ignite/metrics/test_ssim.py [hunk 1]
```diff
@@ -42,38 +43,76 @@ def test_invalid_ssim():
...
+
+    assert isinstance(ignite_ssim, torch.Tensor)
+    assert ignite_ssim.dtype == torch.float64
+    assert ignite_ssim.device == torch.device(device)
+    assert np.allclose(ignite_ssim.numpy(), skimg_ssim, atol=atol)
+
     ssim = SSIM(data_range=1.0, gaussian=False, kernel_size=7, device=device)
-    y_pred = torch.rand(16, 3, 227, 227, device=device)
-    y = y_pred * 0.65
+    y_pred = torch.rand(8, 3, 8, 8, device=device)
```
[the remark this answers is not in the document]
Comment (ydcjeff) [reply tests/ignite/metrics/test_ssim.py:103]: May I know why? using an image just larger than `kernel_size` is also valid.

**Comment by ydcjeff** [comment]
> Currently, to factorize the tests we are implementing _test_something(params) and call them from test_something. Can we do the same thing here ?

Is there some changes to metric tests?

File: tests/ignite/metrics/test_ssim.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vfdev-5) [reply tests/ignite/metrics/test_ssim.py:103]: Actually, maybe it worth to have both checks: large image and small image like you proposes. IMO, test time reduction can be important but we still have to check let's say "common" cases with large images.

[the remark this answers is not in the document]
Comment (ydcjeff) [reply tests/ignite/metrics/test_ssim.py:103]: ok so let's check one small image in one place and large image in one place?

**Comment by vfdev-5** [comment]
> > Currently, to factorize the tests we are implementing _test_something(params) and call them from test_something. Can we do the same thing here ?
> 
> Is there some changes to metric tests?

Maybe, I misunderstood your initial point about pytest parametrize. What I'm talking about now is inside `test_ssim` you setup the metric twice and run almost the same asserts for two test cases. Maybe, this code can be factorized initially with `_test_*` suggestion and on the second step, someone could use pytest parametrize...

File: tests/ignite/metrics/test_ssim.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vfdev-5) [reply tests/ignite/metrics/test_ssim.py:103]: Yes, for example in `test_ssim` we can test small and large images and for other tests keep small images like 28x28.

**Comment by vfdev-5** [comment]
@ydcjeff could you lease address this comment : https://github.com/pytorch/ignite/pull/1467#issuecomment-728103603
I'd like to see `test_ssim` refactored with `_test_ssim` which runs on large images `torch.rand(4, 3, 224, 224, device=device)` and small images `torch.rand(4, 3, 28, 28, device=device)`. Other configs can remain the same. Thanks

**Comment by ydcjeff** [comment]
Just increase the batch to be 12 and 8 since skimage ssim can't compute the batch less than win_size.

**Comment by vfdev-5** [comment]
> Just increase the batch to be 12 and 8 since skimage ssim can't compute the batch less than win_size.

@ydcjeff do you have an idea why ? This seems strange a bit, right ?

**Review by vfdev-5 — APPROVED** [review]
Sklearn limitation is somewhat unrealted, I think.
Thank for the PR !
